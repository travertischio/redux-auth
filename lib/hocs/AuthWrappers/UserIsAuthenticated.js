'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reduxAuthWrapper = require('redux-auth-wrapper');

var _actions = require('../../containers/AuthenticationProvider/actions');

var _selectors = require('../../containers/AuthenticationProvider/selectors');

var _config = require('../../config');

var _config2 = _interopRequireDefault(_config);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UserIsAuthenticated = (0, _reduxAuthWrapper.UserAuthWrapper)({
  authSelector: _selectors.selectUser,
  redirectAction: _actions.redirectActionWithSupportParamInQueryString,
  failureRedirectPath: _config2.default.userIsNotAuthenticatedRedirectPath,
  wrapperDisplayName: 'UserIsAuthenticated'
});

exports.default = UserIsAuthenticated;