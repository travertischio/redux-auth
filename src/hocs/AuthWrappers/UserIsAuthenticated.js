import connectedReduxRedirect from './connectedReduxRedirect';
import { redirectActionWithSupportParamInQueryString } from '../../containers/AuthenticationProvider/actions';
import { selectIsAuthenticated } from '../../containers/AuthenticationProvider/selectors';
import config from '../../config';

const UserIsAuthenticated = connectedReduxRedirect({
  authenticatedSelector: selectIsAuthenticated,
  redirectAction: redirectActionWithSupportParamInQueryString,
  redirectPath: config.userIsNotAuthenticatedRedirectPath,
  wrapperDisplayName: 'UserIsAuthenticated',
});

export default UserIsAuthenticated;
