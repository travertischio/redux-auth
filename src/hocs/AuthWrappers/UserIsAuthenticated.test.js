import React from 'react';
import { createComponentWithRouter } from 'react-unit-testing-utils';
import UserIsAuthenticated from './UserIsAuthenticated';

const PageComponent = () => <div>Page only for authenticated users</div>;
const PageOnlyForAuthenticatedUser = UserIsAuthenticated(PageComponent);

describe('UserIsAuthenticated', () => {
  it('should be defined', () => {
    expect(PageOnlyForAuthenticatedUser).toBeDefined();
  });

  it('should redner PageComponent when user is authenticated', () => {
    const initialState = {
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
        },
      },
    };

    const { wrapper } = createComponentWithRouter(PageOnlyForAuthenticatedUser, initialState);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should not redner PageComponent when user is not authenticated', () => {
    const initialState = {
      auth: {},
    };

    const { wrapper } = createComponentWithRouter(PageOnlyForAuthenticatedUser, initialState);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
