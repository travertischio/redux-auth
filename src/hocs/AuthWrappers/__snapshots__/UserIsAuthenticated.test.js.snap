// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIsAuthenticated should not redner PageComponent when user is not authenticated 1`] = `<main />`;

exports[`UserIsAuthenticated should redner PageComponent when user is authenticated 1`] = `
<main>
  <div>
    Page only for authenticated users
  </div>
</main>
`;
